src/InfoBarWidget.js

```diff
@@ -0,0 +1,122 @@
+var InfoBarWidget = function InfoBarWidget( config ) {
+	var widget = this;
+
+	// Configuration initialization
+	config = config || {};
+
+	// Parent constructor
+	OO.ui.ButtonWidget.parent.call( this, config );
+	OO.ui.mixin.IconElement.call( this, config );
+	OO.ui.mixin.LabelElement.call( this, config );
+	OO.ui.mixin.TitledElement.call( this, config );
+	OO.ui.mixin.FlaggedElement.call( this, config );
+
+	this.closeIcon = new OO.ui.IconWidget( {
+		icon: 'clear',
+		flags: [ 'invert' ],
+		classes: [ 'ext-wwt-infoBarWidget-close' ]
+	} );
+	this.userInfoLabel = new OO.ui.LabelWidget( {
+		label: mw.msg( 'ext-whowrotethat-ready-general' ),
+		classes: [ 'ext-wwt-infoBarWidget-info' ]
+	} );
+	this.$pendingAnimation = $( '<div>' )
+		.addClass( 'ext-wwt-infoBarWidget-spinner' )
+		.append(
+			$( '<div>' )
+				.addClass( 'ext-wwt-infoBarWidget-spinner-bounce' )
+		);
+
+	// Set properties
+	this.setState( config.state || 'pending' );
+	this.setLabel( mw.msg( 'ext-whowrotethat-state-pending' ) );
+
+	// Scroll event
+	$( window ).scroll( () => {
+		var pos = $( '#bodyContent' )[ 0 ].getBoundingClientRect();
+
+		widget.$element.toggleClass(
+			'ext-wwt-infoBarWidget-sticky',
+			pos.top < widget.$element.height()
+		);
```

Comment:
it seems to run fast, but maybe we shouldn't call `.toggleClass` on every pixel that is scrolled. So something like
```
if ( pos.top < widget.$element.height() ) {
	widget.$element.addClass( 'ext-wwt-infoBarWidget-sticky' );
} else {
	widget.$element.removeClass( 'ext-wwt-infoBarWidget-sticky' );
}
```
or maybe jQuery is smart enough to not re-manipulate the DOM (assuming this is jQuery)

Reply:
Yup, I'm using `position: sticky` instead!